AbletonMCP_Remote_Script/handlers/audio.py

```
@@ -0,0 +1,517 @@
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error analyzing audio clip: " + str(e))
+            ctrl.log_message(traceback.format_exc())
+        raise
+
+
+def freeze_track(song, track_index, ctrl=None):
+    """Freeze a track."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)